hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/workflows/FakePreHandleContext.java

```diff
@@ -144,6 +138,41 @@ public Set<Key> requiredNonPayerKeys() {
         return Collections.unmodifiableSet(requiredNonPayerKeys);
     }
 
+    @NonNull
+    @Override
+    public Set<Key> optionalNonPayerKeys() {
+        // @todo('@rbair23') Needs to be implemented
+        return Collections.emptySet();
+    }
+
+    @NonNull
+    @Override
+    public PreHandleContext optionalKey(@NonNull final Key key) {
+        // @todo('@rbair23') Needs to be implemented
+        return this;
+    }
+
+    @NonNull
+    @Override
+    public PreHandleContext optionalKeys(@NonNull final Set<Key> keys) {
+        // @todo('@rbair23') Needs to be implemented
+        return this;
+    }
+
+    @NonNull
+    @Override
+    public Set<Account> optionalHollowAccounts() {
+        // @todo('@rbair23') Needs to be implemented
+        return Collections.emptySet();
+    }
+
+    @NonNull
+    @Override
+    public PreHandleContext optionalSignatureForHollowAccount(@NonNull final Account hollowAccount) {
+        // @todo('@rbair23') Needs to be implemented
+        return this;
+    }
+
```

Comment:
Can you also implement these in this PR? They will follow the same pattern as for the required keys. I will implement the logic in pre-handle to actually call the methods to get the optional hollow accounts and optional keys.

Reply:
Sure, didn't have those available until today, but I emulated those methods now.

Reply:
Done.